adafruit_ads1x15/adafruit_ads1x15.py

```diff
@@ -186,40 +209,24 @@ def _read(self, mux, gain, data_rate, mode):
         config |= self._data_rate_config(data_rate)
         config |= ADS1X15_CONFIG_COMP_QUE_DISABLE  # Disble comparator mode.
         # Send the config value to start the ADC conversion.
-        # Explicitly break the 16-bit value down to a big endian pair of bytes.
-        self.buf[0] = ADS1X15_POINTER_CONFIG
-        self.buf[1] = (config >> 8) & 0xFF
-        self.buf[2] = config & 0xFF
-        with self.i2c_device as i2c:
-            i2c.write(self.buf)
-            # Wait for the ADC sample to finish based on the sample rate plus a
-            # small offset to be sure (0.1 millisecond).
-            time.sleep(1.0/data_rate+0.0001)
-            # Retrieve the result.
-            self.buf[0] = ADS1X15_POINTER_CONVERSION
-            i2c.write(self.buf, end=1, stop=False)
-            i2c.readinto(self.buf, start=1)
-        return self._conversion_value(self.buf[2], self.buf[1])
+        self._write_register(ADS1X15_POINTER_CONFIG, config)
+        # Wait for conversion to complete
+        while not self._conversion_complete():
+            pass
```

Comment:
wouldn't be better to have time.sleep() instead of pass?

Reply:
What's the reason? Easy enough to do, just curious what the trade off is.

Reply:
I suppose it's a good practice from regular Python, where you want to give the control back to the operating system and other applications, instead of running it at 100%.
Perhaps it will also make a difference for SAMD at some point, when we get to the power management options.

Reply:
Gotcha. I figured `pass` took care of that. So is just a zero sleep good enough?
```python
time.sleep(0)
```